components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2881,6 +2881,49 @@ def __str__(self):
         return define
 
 
+class CGCollectJSONAttributesMethod(CGAbstractMethod):
+    """
+    Generate the CollectJSONAttributes method for an interface descriptor
+    """
+    def __init__(self, descriptor, toJSONMethod):
+        args = [Argument('*mut JSContext', 'cx'),
+                Argument('HandleObject', 'obj'),
+                Argument('*const %s' % descriptor.concreteType, 'this'),
+                Argument('&RootedGuard<*mut JSObject>', 'result')]
+        CGAbstractMethod.__init__(self, descriptor, 'CollectJSONAttributes',
+                                  'bool', args, pub=True, unsafe=True)
+        self.toJSONMethod = toJSONMethod
+
+    def definition_body(self):
+        ret = ''
+        interface = self.descriptor.interface
+        for m in interface.members:
```

Comment:
does this deal with inheritance? Should we make it deal with inheritance?

Reply:
Inheritance is supported; not here but at https://github.com/servo/servo/pull/23844/files#diff-60d01595cff328c165842fea9e4ccbc2R3684. Each interface defines its `CollectJSONAttributes` which is called by ToJSON in an ordered way.